Automattic/wp-calypso
# Components: Fix unknown prop warnings in DisconnectJetpackButton

This PR clears JS warnings emitted by the `DisconnectJetpackButton` component - they were caused by unnecessary props being specified to the `Button` component. It also does very basic cleanup to remove an ESLint warning, and always use the `Button` component instead of the `<button>` tag.

Note: this PR is part of the unknown prop warning resolve PR series: #7413.

To test:

* Get this branch going locally.
* Go to `/settings/general/$site`, where `$site` is one of your Jetpack sites.
* Go to `/plugins/$site`, where `$site` is one of your Jetpack sites.
* Go to `/plugins/jetpack/$site`, where `$site` is one of your Jetpack sites.
* In the top 3 cases, verify that there are no JS warnings caused by `DisconnectJetpackButton`, and there are no visual/functional changes to the `DisconnectJetpackButton` component.

/cc @aduth
